### PR TITLE
TOC restructure

### PR DESCRIPTION
**Pull request by jc-berger**
Fixes [#441 top level documentation structure](https://github.com/devfile/api/issues/441). This restructure is to make the documentation TOC more persona focused rather than feature focused. 

Note, I've left some comments to help guide reviewers' attention. 

You can also view the [Gdoc version of the TOC restructure](https://docs.google.com/document/d/1OamiXqtwvWtdHKJRpqJ8HOULUOLlUQV0-uvoZHyp3G8/edit?usp=sharing), which includes comments and conversations regarding the persona focused TOC.

### CODE REVIEW

**Comment by jc-berger** [comment]
For convenience, please see the attached screenshot for how the new TOC would render. Note, I need to correct how `using stacks.` 
Screen Shot 2021-08-23 at 4 25 59 PM

File: docs/modules/user-guide/nav.adoc [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
-** xref:adding-commands-to-a-devfile.adoc[]
-** xref:adding-components-to-a-devfile.adoc[]
-*** xref:adding-kubernetes-component-to-a-devfile.adoc[]
-*** xref:adding-container-component-to-a-devfile.adoc[]
-*** xref:specifying-persistent-storage.adoc[]
-*** xref:limiting-resources-usage.adoc[]
-*** xref:defining-environment-variables.adoc[]
-*** xref:defining-endpoints.adoc[]
-*** xref:defining-kubernetes-resources.adoc[]
+* Developer
```
Comment (jc-berger) [inline docs/modules/user-guide/nav.adoc:3]: Moved developer up first as that is our target persona.

File: docs/modules/user-guide/nav.adoc [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
+** xref:using-stacks.adoc[]
```
Comment (jc-berger) [inline docs/modules/user-guide/nav.adoc:4]: Added a `Using stacks` section before `Authoring stacks` to help the user workflow. Thoughts? 

Also, I'm not sure why the `Using stacks` isn't rendering properly.

File: docs/modules/user-guide/nav.adoc [hunk 3]
```diff
@@ -1,32 +1,36 @@
...
+** xref:migrating-to-devfile-v2.adoc[]
+*** xref:migrating-schema-version.adoc[]
+*** xref:migrating-projects.adoc[]
+*** xref:migrating-components.adoc[]
+*** xref:migrating-plug-ins.adoc[]
+*** xref:migrating-commands.adoc[]
+*** xref:adding-event-bindings.adoc[]
+*** xref:referring-to-a-parent-devfile-in-a-devfile.adoc[]
+
+* Registry Administrator
```
Comment (jc-berger) [inline docs/modules/user-guide/nav.adoc:32]: We can add the rest of the devfile registry docs here once the TOC changes get merged.

**Review by johnmcollier — APPROVED** [review]
Looks good to me 👍

**Pull request merged**